### ISSUE TITLE
ESC causing error

### ISSUE THREAD

**Issue by tolgaduzenli** [snakesilk/react-fullscreen#21]
Same problem is happening for me. It might be because of my structure.  
I want to have fullscreen button inside of fullscreen and toggling for fullscreen or exit.
But the issue is, while screen is full, press the 'ESC', getting below error;

Failed to execute 'requestFullscreen' on 'Element': API can only be initiated by a user gesture.

```  
 <Fullscreen
          enabled={this.state.isFull}
          onChange={isFull => this.setState({ isFull })}
        >
          <Router history={history}>
            <div className="page-container">
              <div className="nav-bar">
                <NavBar history={history} onClickFullScreen={this.handleFullScreen}/>
              </div>
            </div>
          </Router>
        </Fullscreen>
```

_Originally posted by @tolgaduzenli in https://github.com/snakesilk/react-fullscreen/issues/14#issuecomment-483307856_

**Comment by petrovicstefanrs** [comment]
Having the exact same issue. It would be great to know if this is due to the button controlling fullscreen being inside the Fullscreen component.

**Comment by matinfo** [comment]
If you use React `useState()`, aka: `const [ isFull, setIsFull ] = useState(false);`, change to `onChange={isFull => setIsFull(isFull)}`

**Comment by alexmac131** [comment]
I too am having the same issue, I have nested full screens - so the outside works fine - and the inner full screen throws the non fatal error when I hit ESC

**Comment by matinfo** [comment]
This work:
```
const MainLayout = (Component) => (props) => {
  const [ isFull, setIsFull ] = useState(false);
  return <Fullscreen
      enabled={isFull}
      onChange={(isFull) => setIsFull(isFull)}>`
         .... rest of the layout ....
        <MainFooter setIsFull={setIsFull} isFull={isFull} />
      </Layout>
    </Fullscreen>;
};
```

and footer look like this:

```
const MainFooter = ({ isFull, setIsFull }) => {
  return (
    <Footer className="footer">
    ... rest of footer ...
         {
            isFull 
             ? <Button size="small" type="link" onClick={() => setIsFull(false)}>exist fullscreen</Button>
              : <Button size="small" type="link" onClick={() => setIsFull(true)}>enter fullscreen</Button>
          }
    </Footer>;
};
```

Hope that help.

**Comment by alexmac131** [comment]
@matinfo  this is great but for legacy reasons , I can't use react hooks, is there another means without hooks?

**Comment by vianneyguesdon** [comment]
Issue still open. ESC causing a problem

**Comment by matinfo** [comment]
Not for me: https://meqvm.csb.app/

**Comment by djordjesavanovic** [comment]
I have the same problem. Even if I try to wrap it in a div and handle it with a keypress or keydown. This happens only in Chrome. In Safari, it's including extra elements in fullscreen.

**Comment by pomle** [comment]
So not sure what the common problem is here, but having multiple full screen elements connected to a single state will cause problems in this version. 

I recommend everyone to try this:
https://github.com/snakesilk/react-fullscreen/issues/27#issuecomment-657026335

**Comment by pomle** [comment]
This problem seems to have been solved by 0.3.0 and followed up here https://github.com/snakesilk/react-fullscreen/issues/27
